sys/net/gnrc/routing/rpl/gnrc_rpl_control_messages.c

```diff
@@ -97,28 +148,30 @@ void gnrc_rpl_send_DIO(gnrc_rpl_instance_t *inst, ipv6_addr_t *destination)
     }
 
     gnrc_rpl_dodag_t *dodag = &inst->dodag;
-    gnrc_pktsnip_t *pkt;
-    icmpv6_hdr_t *icmp;
+    gnrc_pktsnip_t *pkt = NULL, *tmp = NULL;
     gnrc_rpl_dio_t *dio;
-    uint8_t *pos;
-    int size = sizeof(icmpv6_hdr_t) + sizeof(gnrc_rpl_dio_t);
 
-    if (dodag->dodag_conf_requested) {
-        size += sizeof(gnrc_rpl_opt_dodag_conf_t);
+    if (dodag->prefix_info_requested) {
+        if ((pkt = _dio_prefix_info_build(pkt, dodag)) == NULL) {
+            return;
```

Comment:
I know, but RPL wrongly assumes now it successfully sent a DIO, and in consequence it will be eventually deleted from the parent sets of its children if this situation sustains